templates/htan-synapse-bucket.yaml

```diff
@@ -107,20 +107,21 @@ Resources:
               - s3:PutObject
               - s3:PutObjectAcl
             Resource: !Sub "${S3Bucket.Arn}/*"
-          - Effect: Deny
-            Principal: "*"
-            Action: s3:PutObject
-            Resource: !Sub "${S3Bucket.Arn}/*"
-            Condition:
-              StringNotEqualsIgnoreCase:
-                s3:x-amz-grant-full-control: !Sub "id=${SynapseCanonicalId},id=${S3CanonicalUserId}"
-              StringNotEquals:
-                s3:x-amz-grant-full-control: !Sub "id=${SynapseCanonicalId},id=${S3CanonicalUserId}"
+#          - Effect: Deny
+#            Principal: "*"
+#            Action: s3:PutObject
+#            Resource: !Sub "${S3Bucket.Arn}/*"
+#            Condition:
+#              StringNotEqualsIgnoreCase:
+#                s3:x-amz-grant-full-control: !Sub "id=${SynapseCanonicalId},id=${S3CanonicalUserId}"
+#              StringNotEquals:
+#                s3:x-amz-grant-full-control: !Sub "id=${SynapseCanonicalId},id=${S3CanonicalUserId}"
```

Comment:
i'm not opposed to making that change @clarisse-lau however I'm not sure how your suggestion provides more flexibility?


I'm not sure what your suggestion applies to @xschildw?  is it meant as an alternative to the structure of this `Condition` clause? or is your suggestion meant as a possible work around for the `Deny` policy issue?

Reply:
My proposal was re: comment to have the Deny policy apply to all principals except one role (the service role that writes owner.txt). To do that we look at 'aws:userid' and exclude everything but roleID:* (so the role would not be denied).

Reply:
@zaro0508 this came up because when we were testing (no spaces) one of us had spaces in between the ids and it was not accepted, I think what is in between `" "` needs to be an exact string match.

Reply:
implemented @clarisse-lau and @xschildw suggestions.  Does that look correct @xschildw ?